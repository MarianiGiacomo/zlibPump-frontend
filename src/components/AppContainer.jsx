import React from 'react';
import { render } from 'react-dom';
import CssBaseline from '@material-ui/core/CssBaseline';
import { ThemeProvider as MuiThemeProvider } from '@material-ui/core/styles';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import AppGrid from './presentational/AppGrid';
import reducers from '../reducers';
import { theme } from '../styles';

const store = createStore(reducers);

export default function AppContainer() {
  return (
    <Provider store={store}>
      <MuiThemeProvider theme={theme}>
        <React.Fragment>
          <CssBaseline />
          <AppGrid />
        </React.Fragment>
      </MuiThemeProvider>
    </Provider>
  );
}

const wrapper = document.getElementById('AppContainer');
if (wrapper) {
  render(<AppContainer />, wrapper);
}
