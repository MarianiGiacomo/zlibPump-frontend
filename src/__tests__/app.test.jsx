import '@babel/polyfill';
import React from 'react';
import {
  render,
  fireEvent,
  cleanup,
  waitForElement,
} from '@testing-library/react';
import AppContainer from '../components/AppContainer';

afterEach(() => {
  cleanup();
  console.error.mockClear();
  console.warn.mockClear();
});


console.error = jest.fn();
console.warn = jest.fn();

test('Test snapshot', () => {
  const { container } = render(<AppContainer />);
  expect(container.firstChild).toMatchSnapshot();
});

test('Renders without errors or warnings', () => {
  render(<AppContainer />);
  expect(console.error).not.toHaveBeenCalled();
  expect(console.warn).not.toHaveBeenCalled();
});

test('Page renders the right amount of Form divs', () => {
  const { getAllByTestId } = render(<AppContainer />);

  expect(getAllByTestId('form-div').length).toBe(4);
});

test('Textfields render with the right id', () => {
  const { getByLabelText } = render(<AppContainer />);

  expect(getByLabelText('Inflated ASCII').id).toBe('inflatedAscii');
  expect(getByLabelText('Inflated Hex').id).toBe('inflatedHex');
  expect(getByLabelText('Deflated Hex').id).toBe('deflatedHex');
  expect(getByLabelText('Deflated base64').id).toBe('deflatedBase64');
});

test('Handling of empthy values', () => {
  const { getByTestId } = render(<AppContainer />);
  const deflateAscii = getByTestId('button-inflatedAscii');
  const deflateHex = getByTestId('button-inflatedHex');
  const inflateHex = getByTestId('button-deflatedHex');
  const inflateBase64 = getByTestId('button-deflatedBase64');

  fireEvent.click(deflateAscii);
  expect(document.getElementById('inflatedAscii-helper-text').textContent).toBe('Please insert some text');
  fireEvent.click(deflateHex);
  expect(document.getElementById('inflatedHex-helper-text').textContent).toBe('Please insert some text');
  fireEvent.click(inflateHex);
  expect(document.getElementById('deflatedHex-helper-text').textContent).toBe('Please insert some text');
  fireEvent.click(inflateBase64);
  expect(document.getElementById('deflatedBase64-helper-text').textContent).toBe('Please insert some text');
});

test('Call with valid value, update textfields with return values ', async () => {
  const { getByLabelText, getByTestId, getByText } = render(<AppContainer />);

  fireEvent.change(getByLabelText('Inflated ASCII'), { target: { value: 'Valid value' } });
  fireEvent.click(getByTestId('button-inflatedAscii'));

  // Wait for one of firebase mock function return values defined in /__mocks__/firebase.js
  await waitForElement(() => getByText('inflatedHex'));

  expect(getByLabelText('Inflated Hex').textContent).toBe('inflatedHex');
  expect(getByLabelText('Deflated Hex').textContent).toBe('deflatedHex');
  expect(getByLabelText('Deflated base64').textContent).toBe('deflatedBase64');
  expect(getByLabelText('Inflated ASCII').textContent).toBe('inflatedAscii');
});

test('Call with invalid value, open modal with error message and close modal', async () => {
  const {
    getByTestId,
    getByLabelText,
    getByText,
    queryByTestId,
  } = render(<AppContainer />);

  // Test open modal with error message when invalid value returns an error
  fireEvent.change(getByLabelText('Inflated Hex'), { target: { value: 'Invalid value' } });
  fireEvent.click(getByTestId('button-inflatedHex'));

  await waitForElement(() => getByText('Error message'));
  expect(getByTestId('modal')).toBeTruthy();

  // Test closing modal
  fireEvent.click(getByTestId('button-modal'));
  expect(queryByTestId('modal')).toBe(null);
});
