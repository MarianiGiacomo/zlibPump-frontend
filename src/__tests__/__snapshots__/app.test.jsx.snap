// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snapshot 1`] = `
<div
  class="AppGrid-root-1"
>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-10"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom MuiTypography-alignCenter"
      >
        ZlibPump
      </h6>
      <h6
        class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom MuiTypography-alignCenter"
      >
        Inflate and deflate strings with zlib
      </h6>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
    >
      <div
        class="InflateBox-root-141"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-8"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation1 InflateBox-paper-142 MuiPaper-rounded"
            >
              <div
                data-testid="form-div"
              >
                <form
                  class="Form-form-177"
                >
                  <div
                    class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                      data-shrink="false"
                      for="deflatedHex"
                      id="deflatedHex-label"
                    >
                      Deflated Hex
                    </label>
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
                    >
                      <textarea
                        aria-invalid="false"
                        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                        id="deflatedHex"
                        rows="1"
                        style="height: 0px; overflow: hidden;"
                      />
                      <textarea
                        aria-hidden="true"
                        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                        readonly=""
                        style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); width: 100%;"
                        tabindex="-1"
                      />
                      <fieldset
                        aria-hidden="true"
                        class="PrivateNotchedOutline-root-242 MuiOutlinedInput-notchedOutline"
                        style="padding-left: 8px;"
                      >
                        <legend
                          class="PrivateNotchedOutline-legend-243"
                          style="width: 0.01px;"
                        >
                          <span>
                            ​
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                    
                  </div>
                  <div
                    class="Form-centering-178"
                  >
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained Form-button-179"
                      data-testid="button-deflatedHex"
                      tabindex="0"
                      type="submit"
                    >
                      <span
                        class="MuiButton-label"
                      >
                        Inflate
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </form>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation1 InflateBox-paper-142 MuiPaper-rounded"
            >
              <div
                data-testid="form-div"
              >
                <form
                  class="Form-form-177"
                >
                  <div
                    class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                      data-shrink="false"
                      for="deflatedBase64"
                      id="deflatedBase64-label"
                    >
                      Deflated base64
                    </label>
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
                    >
                      <textarea
                        aria-invalid="false"
                        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                        id="deflatedBase64"
                        rows="1"
                        style="height: 0px; overflow: hidden;"
                      />
                      <textarea
                        aria-hidden="true"
                        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                        readonly=""
                        style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); width: 100%;"
                        tabindex="-1"
                      />
                      <fieldset
                        aria-hidden="true"
                        class="PrivateNotchedOutline-root-242 MuiOutlinedInput-notchedOutline"
                        style="padding-left: 8px;"
                      >
                        <legend
                          class="PrivateNotchedOutline-legend-243"
                          style="width: 0.01px;"
                        >
                          <span>
                            ​
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                    
                  </div>
                  <div
                    class="Form-centering-178"
                  >
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained Form-button-179"
                      data-testid="button-deflatedBase64"
                      tabindex="0"
                      type="submit"
                    >
                      <span
                        class="MuiButton-label"
                      >
                        Inflate
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
    >
      <div
        class="DeflateBox-root-275"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-8"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation1 DeflateBox-paper-276 MuiPaper-rounded"
            >
              <div
                data-testid="form-div"
              >
                <form
                  class="Form-form-177"
                >
                  <div
                    class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                      data-shrink="false"
                      for="inflatedAscii"
                      id="inflatedAscii-label"
                    >
                      Inflated ASCII
                    </label>
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
                    >
                      <textarea
                        aria-invalid="false"
                        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                        id="inflatedAscii"
                        rows="1"
                        style="height: 0px; overflow: hidden;"
                      />
                      <textarea
                        aria-hidden="true"
                        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                        readonly=""
                        style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); width: 100%;"
                        tabindex="-1"
                      />
                      <fieldset
                        aria-hidden="true"
                        class="PrivateNotchedOutline-root-242 MuiOutlinedInput-notchedOutline"
                        style="padding-left: 8px;"
                      >
                        <legend
                          class="PrivateNotchedOutline-legend-243"
                          style="width: 0.01px;"
                        >
                          <span>
                            ​
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                    
                  </div>
                  <div
                    class="Form-centering-178"
                  >
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained Form-button-179"
                      data-testid="button-inflatedAscii"
                      tabindex="0"
                      type="submit"
                    >
                      <span
                        class="MuiButton-label"
                      >
                        Deflate
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </form>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation1 DeflateBox-paper-276 MuiPaper-rounded"
            >
              <div
                data-testid="form-div"
              >
                <form
                  class="Form-form-177"
                >
                  <div
                    class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                      data-shrink="false"
                      for="inflatedHex"
                      id="inflatedHex-label"
                    >
                      Inflated Hex
                    </label>
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
                    >
                      <textarea
                        aria-invalid="false"
                        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                        id="inflatedHex"
                        rows="1"
                        style="height: 0px; overflow: hidden;"
                      />
                      <textarea
                        aria-hidden="true"
                        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                        readonly=""
                        style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); width: 100%;"
                        tabindex="-1"
                      />
                      <fieldset
                        aria-hidden="true"
                        class="PrivateNotchedOutline-root-242 MuiOutlinedInput-notchedOutline"
                        style="padding-left: 8px;"
                      >
                        <legend
                          class="PrivateNotchedOutline-legend-243"
                          style="width: 0.01px;"
                        >
                          <span>
                            ​
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                    
                  </div>
                  <div
                    class="Form-centering-178"
                  >
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained Form-button-179"
                      data-testid="button-inflatedHex"
                      tabindex="0"
                      type="submit"
                    >
                      <span
                        class="MuiButton-label"
                      >
                        Deflate
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <h6
    class="MuiTypography-root AppGrid-footer-6 MuiTypography-subtitle2 MuiTypography-gutterBottom MuiTypography-alignCenter"
  >
    <a
      href="https://twitter.com/jamomani"
      rel="noreferrer noopener"
      target="_blank"
    >
      By @jamomani
    </a>
  </h6>
</div>
`;
